packages/ember-glimmer/lib/helpers/get.ts

```diff
@@ -105,26 +113,13 @@ class GetHelperReference extends CachedReference {
 
   compute() {
     let { lastPath, innerReference, innerTag } = this;
-
-    let path = this.lastPath = this.pathReference.value();
+    let path = this.pathReference.value();
 
     if (path !== lastPath) {
-      if (path !== undefined && path !== null && path !== '') {
-        let pathType = typeof path;
-
-        if (pathType === 'string' && path.indexOf('.') > -1) {
-          innerReference = referenceFromParts(this.sourceReference, path.split('.'));
-        } else {
-          innerReference = this.sourceReference.get(path);
-        }
-
-        innerTag.inner.update(innerReference.tag);
-      } else {
-        innerReference = NULL_REFERENCE;
-        innerTag.inner.update(CONSTANT_TAG);
-      }
-
+      innerReference = referenceFromPath(this.sourceReference, path);
+      innerTag.inner.update(innerReference.tag);
       this.innerReference = innerReference;
+      this.lastPath = path;
```

Comment:
FWIW, I think this fixed another related bug (we previously _always_ created a new reference even if the path was not changed)...